File: android/app/src/main/java/io/ionic/starter/UWBPlugin.kt
package io.ionic.starter

import com.getcapacitor.BridgeActivity
import com.getcapacitor.Plugin
import com.getcapacitor.annotation.CapacitorPlugin
import com.getcapacitor.JSObject
import com.getcapacitor.PluginCall
import com.getcapacitor.PluginMethod

@CapacitorPlugin(name = "UWBPlugin")
class UWBPlugin : Plugin() {
    private var uwbManager: UWBManager? = null

    override fun load() {
        uwbManager = UWBManager(context)
    }

    @PluginMethod
    fun startScanning(call: PluginCall) {
        uwbManager?.startScanning()
        val result = JSObject()
        result.put("status", "scanning started")
        call.resolve(result)
    }
}

File: ios/UWBPlugin.swift
import Capacitor

@objc(UWBPlugin)
public class UWBPlugin: CAPPlugin {
    let uwbManager = UWBManager()

    @objc func startScanning(_ call: CAPPluginCall) {
        DispatchQueue.main.async {
            self.uwbManager.startUWBInteraction()
            call.resolve(["status": "scanning started"])
        }
    }
}
